src/Sema.zig

```diff
@@ -27010,8 +27014,8 @@ fn explainWhyTypeIsNotPacked(
         .ErrorSet,
         .AnyFrame,
         .Optional,
-        .Array,
         => try sema.errNote(src_loc, msg, "type has no guaranteed in-memory representation", .{}),
+        .Array => try sema.errNote(src_loc, msg, "packed fields are ordered according to machine endianness, array elements are not", .{}),
```

Comment:
Error message sounds more like a warning and should not explain how packed structs work.
A better error message would be something like "arrays are not allowed in packed structs"

Reply:
The reason for it not being allowed is rather subtle, and I think explaining it helps the user understand whether they want to solve the error by giving up packed-ness or array-ness. As far as I understand, the reason #12547 is still open is that the status quo error message saying only "this isn't allowed" is not useful to new users.